psl-core/src/main/java/org/linqs/psl/model/predicate/Predicate.java

```diff
@@ -107,6 +110,28 @@ public static Predicate get(String name)  {
         return predicates.get(name.toUpperCase());
     }
 
+    @Override
+    public int hashCode() {
+        return hashcode;
+    }
+
+    @Override
+    public boolean equals(Object oth) {
+        if (oth == this) {
+            return true;
+        }
+
+        if (!(oth instanceof Predicate)) {
+            return false;
+        }
+
+        Predicate other = (Predicate)oth;
+
+        // First check the hashcode to reduce the time we have to do a deepEquals() on the arguments.
+        // Note that the hashcode is not perfect, but provides a quick insurance on inequality.
```

Comment:
Remove this comment here.
It makes more sense to have in Atom, we don't need to carry it over to every equals() method.